components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/utils/HikariDataSourceUtils.java

```diff
@@ -47,6 +48,12 @@ public static HikariConfig buildConfiguration(String xmlConfiguration) throws Da
             config.setMaximumPoolSize(configuration.getMaximumPoolSize());
             config.setMinimumIdle(configuration.getMinimumIdle());
             config.setAutoCommit(configuration.isAutoCommit());
+            if (configuration.getDatabaseProps() != null && !configuration.getDatabaseProps().isEmpty()) {
+                Properties properties = new Properties();
+                configuration.getDatabaseProps()
+                        .forEach(property -> properties.setProperty(property.getName(), property.getValue()));
```

Comment:
better to move . sign to the upper part of the statement to imply continuation. But your approach could be a different standard, better check on this.